src/app/notes/note.model.ts

```diff
@@ -0,0 +1,15 @@
+export class Note {
+    public title: string;
+    public data: string;
+    public description: string;
+    public id: number;
+    public imagePath: string;
+
+    constructor(title:string, data: string, desc: string, id: number, imgPath?: string) {
```

Comment:
duzo argumentow funkcji to anty wzorzec ,
mozna zrobic to tak 

> constructor(options?:{title:string, data: string, desc: string, id: number, imgPath?: string} = {}){
this.title = options.title || null;
}

Reply:
czy jeśli zrobie to w tej wersji czyli options?: ... to czy imgPath?: string  będzie juz bez "?" ?

Reply:
teoretycznie to luzuje warunki tworzenia obiektu bo mozesz odlic  new Note() bez niczego i wtedy masz Note ze structura id,imagePath desc . bo masz wartosc domysla {} .

na plus ze unikasz sytuacji gdzie nie masz jakis pol i dostajesz undefined.

na minus ze IDE nie wali na czerwono gdy cos takie chcesz zrobic.

Ja bym to stosowal jak cos to prywatny konstructor i metoda statyczna create() ktora tworzy Ci wedlug jakis tam zasad obiekt.

Reply:
mozesz napisac statyczna metode createDefault() { 
// jakas logika
return new Notes(//jakies parametry)}

a obok create() { 
// jakas inna logika
return new Notes(//jakies inne parametry)}